### DISCUSSION TITLE
Vertical Datum Handling

### DISCUSSION THREAD

**Discussion by msweier** [USACE/cwms-data-api#853] [Database to API User data mapping]
I see that for some elevation timeseries, a vertical datum conversion is included. For example,
```
curl -X 'GET' \
  'https://cwms-data.usace.army.mil/cwms-data/timeseries?name=TraverseRES_Dam.Elev.Inst.15Minutes.0.rev-MSL1912&office=MVP' \
  -H 'accept: application/json;version=2'
```
yeilds the timeseries data and then at the end:
```
  "vertical-datum-info": {
    "office": "MVP",
    "unit": "ft",
    "location": "TraverseRES_Dam",
    "native-datum": "OTHER",
    "elevation": 900,
    "offsets": [
      {
        "estimate": false,
        "to-datum": "NAVD-88",
        "value": 0.31
      },
      {
        "estimate": true,
        "to-datum": "NGVD-29",
        "value": -0.3903
      }
```
What is interesting is I cannot find where this conversion is defined in the CWMS database. Where are the offsets stored?  Are these associated w/ just the timeseries or location?

I don't think we have anything like stored in the database. The conversions appear to be correct or pretty close even with the oddball MSL 1912.  What will the future look like when the new GEOID2022 is required? Can stage timeseries support this for multiple conversions or is this just for Elev timeseries? Any plans to allow managing in CWMS-VUE? This all kind of ties into this discourse post I had https://discourse.hecdev.net/t/datum-management-in-cwms-database/3361/3

**Comment by msweier** [comment]
@MikeNeilson and Mike Perryman, penny for your thoughts?

**Comment by MikeNeilson** [comment]
The conversions are stored per location. It uses a specific algorithm to calculate what it should be. Like you said it's generally quite close.

I'm not super aware of the future datum change, but I would assume it would work the same way, calculate by default, allow setting something else.... can't remember how to do that though.

**Reply by msweier** [reply]
That's interesting. The example I picked is in MSL 1912 and classified as "Local" for the vertical datum. There aren't any conversion algos I know of out there for it other than surveying on a case by case basis. So I'm thinking someone entered this info in somewhere. @Enovotny can CWMS-VUE be updated to view/edit this? Is there a SQL query that can let me see what's in there now?

![image](https://github.com/user-attachments/assets/22d4fdf7-a2ce-41d2-804d-be886d04e82c)

Here's the info on the much delayed 2022 datum. https://geodesy.noaa.gov/datums/newdatums/index.shtml It will come....eventually....it's a GPS based datum so in theory should help reduce the shifting conversion nonsense we have gone through.
